aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/handlers/downstream/SqlHandler.java

```diff
@@ -56,8 +59,13 @@ public void handleRequest(Event event) {
             return;
         }
 
+        // If the query string wasn't provided by current DiSCo event, check the preparedMap cache
```

Comment:
Is it a real case that queryString is null and at the same time StatementMap contains the statement key and value?

Reply:
The cache insertion should always happen for a PreparedStatement before we get to this point, I explained it more here: https://github.com/aws/aws-xray-java-agent/pull/92/files#r604483784

That being said, if the cache is somehow empty, then we simply won't record the SQL query string in the subsegment and there's not much we can do.

Reply:
I mean it seems here `requestEvent.getOperation()` is always same with the value in cache because the cache is set by `requestEvent.getOperation()` before get to this point? If yes, can we remove line 49 and get value from Cache directly?

Reply:
Ah I see. Although they are both handling the same event type, the event sources (and the data available) are different in each case depending on the parameters of the captured call.

So a typical customer setup is like this:

```java
// Emits a Disco event with operation set to the first argument. This populates the cache.
PreparedStatement stmt = conn.prepareStatement("SELECT * FROM my_table");  

// ...

// Emits a Disco event with no operation because it's unavailable in args. Instead, we read the query from the cache.
stmt.execute();
```

Reply:
I see, thanks for explanation.